src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/IHeartbeatProvider.cs

```diff
@@ -4,10 +4,22 @@
     using System.Collections.Generic;
     using Microsoft.ApplicationInsights.Extensibility;
 
-    internal interface IHeartbeatProvider : IHeartbeatPropertyManager, IDisposable
+    internal interface IHeartbeatProvider : IDisposable
```

Comment:
Please add comments about all the interfaces - so it'll become clearer the purpose of each of them.

Reply:
Done. Also, I was able to remove some of the redundant duplication.